### DISCUSSION TITLE
Cannot get slack webhook to work

### DISCUSSION THREAD

**Discussion by Ic3Nine** [jgyates/genmon#1069] [Questions] [answered]
### Checklist

- [X] I made sure that there are *no existing issues or discussions* - [open issues](https://github.com/jgyates/genmon/issues),[closed issues](https://github.com/jgyates/genmon/issues?q=is%3Aissue+is%3Aclosed),[open discussion](https://github.com/jgyates/genmon/discussions), [closed discussion](https://github.com/jgyates/genmon/discussions/categories/questions?discussions_q=is%3Aclosed) - which may answer my question or that I could contribute my information to.
- [X] I have read the [Known Issues Wiki Page](https://github.com/jgyates/genmon/wiki/Appendix-D-Known-Issues) and my question / issue is not listed.
- [X] I have provided the version of the software and the operating system in my post.
- [X] If I have an issue with genmon not communicating with the controller, I have read and followed the [serial troubleshooting guide](https://github.com/jgyates/genmon/wiki/3.6---Serial-Troubleshooting).

### Version

Genmon
Version V1.19.02
running on rpi4

### Have you submitted your log files from the About page? Note that outbound email must be working for logs to be submitted.

I have not submitted logs

### What is/are your question(s)?

I have configured the slack addon and enabled all the notification options.  I don't see any messages being posted in the channel.  Am I missing something obvious?
![image](https://github.com/jgyates/genmon/assets/11322798/fe881180-59e6-4fb8-8b3c-6fdee8111c01)

**Comment by jgyates** [comment]
I have not received your log files. Outbound email must be setup with genmon on the settings page to successfully submit logs.

The log file /var/log/genslack.log may contain more info to your issue.

**Reply by Ic3Nine** [reply]
Apologies, i don't have email setup but the error in genslack is this
2024-03-05 11:00:05,364 : Error: invalid title_link setting

Also, it looks like the title_link is auto populated depending on how you access genmon.
![image](https://github.com/jgyates/genmon/assets/11322798/1c4f1ae6-0a2c-4879-bfe4-b9585bc3dcf8)

**Comment by Ic3Nine** [comment, marked as answer]
ok so it looks like it doesn't like the port.  If i take out :8000 it gives a green check and am now getting messages in slack channel.  Can't click on the title but at least I got it working.

![image](https://github.com/jgyates/genmon/assets/11322798/0f03fbf1-46d0-4749-9922-ad7896fe686e)

Edit - I went into settings and changed the web port to 80; don't have anything else running on the pi so no need for a custom port.

**Comment by jgyates** [comment]
try changing it back to 8000 and see if that makes a difference. The problem may be that the javacode puts the base URL in the title link box as the default, but if that value never changes it does not write it out (which is a bug). Now that you have changed it, you can change it back to 8000 and it should work (i think). I can not get it to fail on my testing so this is my best guess. If that works for you I can change the defaut title to blank and that should fix it for everyone.

Also, the message about a valid URL is just a warning, this was canned code that checks for a value URL, but I can add a check for IP address too.

**Reply by Ic3Nine** [reply]
You are indeed correct.  Changed it back to 8000 in slack config and it still works.  The ole "i'm going to show a value here but won't actually write it to config" bug.  heh.  thanks for your help.
Now if some better geek than myself can do a discord add-on, business!

**Reply by jgyates** [reply]
I checked in an update that will not longer add the current genmon web URL as the default. Instead it will be blank if it has not been initialized, which matches the default of the genslack.conf file. this should correct the issue. I will also check in an update soon for the URL validation regex that will allow both IP and FQDN type addresses with optional port numbers.

**Reply by jgyates** [reply]
I updated the regex to be a bit less strict.
